shuup/gdpr/utils.py

```diff
@@ -171,9 +190,32 @@ def create_user_consent_for_all_documents(shop, user):
     """
     from shuup.gdpr.models import GDPRUserConsent, GDPRSettings
 
-    if not GDPRSettings.get_for_shop(shop).enabled or is_documents_consent_in_sync(shop, user):
+    gdpr_settings = GDPRSettings.get_for_shop(shop)
+    if not gdpr_settings.enabled or is_documents_consent_in_sync(shop, user):
         return
 
-    from shuup.simple_cms.models import Page, PageType
-    consent_documents = Page.objects.visible(shop).filter(page_type=PageType.REVISIONED)
+    consent_documents = get_active_consent_pages(shop)
     GDPRUserConsent.create_for_user(user, shop, consent_documents)
+
+
+def get_possible_consent_pages(shop):
+    return Page.objects.filter(shop=shop, deleted=False, page_type=PageType.REVISIONED)
+
+
+def get_active_consent_pages(shop):
+    from shuup.gdpr.models import GDPRSettings
+    gdpr_settings = GDPRSettings.get_for_shop(shop)
+    if not gdpr_settings.enabled:
+        return False  # nothing to do.
+    ids = [page.id for page in gdpr_settings.consent_pages.all()]
+    if gdpr_settings.privacy_policy_page:
+        ids.append(gdpr_settings.privacy_policy_page.pk)
+    return Page.objects.filter(pk__in=ids)
```

Comment:
make ids a `set()` to prevent duplicates or use `distinct()` since one can select the document in privacy_policy and add it on documents